### PR TITLE
Various clean-ups and configuration improvements

### PR DESCRIPTION
**Pull request by iangmaia**
## What does it do?

It contains the changes implemented in the following Pull Requests:
- https://github.com/wordpress-mobile/release-toolkit/pull/445
- https://github.com/wordpress-mobile/release-toolkit/pull/447
- https://github.com/wordpress-mobile/release-toolkit/pull/450
- https://github.com/wordpress-mobile/release-toolkit/pull/451
- https://github.com/wordpress-mobile/release-toolkit/pull/490

## Checklist before requesting a review

- [x] Run `bundle exec rubocop` to test for code style violations and recommendations
- [x] Add Unit Tests (aka `specs/*_spec.rb`) if applicable
- [x] Run `bundle exec rspec` to run the whole test suite and ensure all your tests pass
- [x] Make sure you added an entry in [the `CHANGELOG.md` file](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/CHANGELOG.md#trunk) to describe your changes under the approprioate existing `###` subsection of the existing `## Trunk` section.

### CODE REVIEW

**Comment by iangmaia** [comment]
@wordpress-mobile/apps-infrastructure -- during my trial, at some point we've created this "base branch" for merging some of the PRs I created so that we could review and approve them quicker, because back then we were not yet ready for some of the breaking changes to land and block a major release.
Even though this PR already contains the PRs mentioned in the description that have been reviewed before, it might be helpful to have more people (especially those who were not involved in the trial) take a quick look at it before we merge the changes.

File: spec/ios_get_build_number_spec.rb [hunk 1]
```diff
@@ -13,14 +13,14 @@
       expect_build_number(xcconfig_mock_content: xcconfig_mock_content, expected_build_number: '1940')
     end
 
-    it 'parses an xcconfig file with keys with spaces and returns a nil build number' do
+    it 'parses an xcconfig file with keys with spaces and returns the correct build number' do
```
Comment (iangmaia) [inline spec/ios_get_build_number_spec.rb:16]: This test was verifying something else given we didn't use xcodeproj for parsing the `.xcconfig` file (see [this comment](https://github.com/wordpress-mobile/release-toolkit/pull/458#discussion_r1178582147) from @spencertransier).

Comment (spencertransier) [reply spec/ios_get_build_number_spec.rb:16]: Awesome, thanks for updating this test @iangmaia!

File: lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_version_helper.rb [hunk 2]
```diff
@@ -15,12 +17,33 @@ module VersionHelper
...
+        #
+        # @return [String] The public-facing version number, extracted from the VERSION_LONG entry of the xcconfig file.
+        #         - If this version is a hotfix (more than 2 parts and 3rd part is non-zero), returns the "X.Y.Z" formatted string
+        #         - Otherwise (not a hotfix / 3rd part of version is 0), returns "X.Y" formatted version number
+        #
+        # @deprecated This method is going to be removed soon due to it's dependency on `ENV['PUBLIC_CONFIG_FILE']` via `get_build_version`.
+        #
         def self.get_public_version
-          version = get_build_version
+          version = get_build_version()
```
Comment (spencertransier) [inline lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_version_helper.rb]: I was wondering why Rubocop didn't flag this but looks like that's still in the `rubocop-todo.yml` for this repo: https://github.com/wordpress-mobile/release-toolkit/blob/eeb93a90a037c33550c691c2595f28a70b9b9f3a/.rubocop_todo.yml#L256-L260

**Review by spencertransier — APPROVED** [review]
These changes all look great @iangmaia. Thanks for your awesome work during your trial! 👏

File: lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_version_helper.rb (hunk 2 above)
Comment (iangmaia) [reply lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_version_helper.rb]: After rebasing, now the style errors are being flagged 🎉 
(update: it flagged several times but on different files 😌 )

**Pull request merged**